libraries/freertos_plus/standard/freertos_plus_tcp/source/FreeRTOS_DNS.c

```diff
@@ -1308,6 +1308,9 @@ TickType_t xTimeoutTime = pdMS_TO_TICKS( 200 );
 		pxUDPHeader->usLength			   = FreeRTOS_htons( lNetLength + ipSIZE_OF_UDP_HEADER );
 		vFlip_16( pxUDPPacket->xUDPHeader.usSourcePort, pxUDPPacket->xUDPHeader.usDestinationPort );
 
+		/* Important: tell NIC driver how many bytes must be sent */
+		pxNetworkBuffer->xDataLength = ( size_t ) ( lNetLength + ipSIZE_OF_IPv4_HEADER + ipSIZE_OF_UDP_HEADER + ipSIZE_OF_ETH_HEADER );
```

Comment:
As in the PR for usGenerateProtocolChecksum, I have to ask; is this function only used for ethernet frames, and never with any other layer 2 type?

Reply:
It is used for ethernet frames only, just like all other FreeRTOS+TCP functions.